hack/generate_tests.py

```diff
@@ -2,26 +2,54 @@
 """Regenerate tests for only the files that have changed."""
 
 import argparse
+import jinja2
 import logging
 import os
+import shutil
 import subprocess
 
-TOP_LEVEL_EXCLUDES = ["docs", "hack", "tests", "kfdef"]
+TOP_LEVEL_EXCLUDES = ["docs", "hack", "kfdef", "stacks", "tests"]
```

Comment:
Would kfdef need separate handling for testing through this mechanism?
It would be great if we can have kfdefs built and tested through this harness.

Reply:
I think partially and I think that relates to kubeflow/kfctl#57.

If we think of KFDef and kfctl as a shorthand for generating a much larger more verbose set of declarative configs then we can follow a similar approach of testing that KFDef generates the expected configs. i.e. we could run `kfctl build` (but not `kfctl apply` and compare the actual to expected configurations. 

So the general approach would work but the mechanics would be different (i.e. we wouldn't do kustomize build).

Reply:
This makes sense.

Another aspect I was thinking about testing here is our generation of versioned kfdefs through kustomize - though this will be a narrower approach than your suggestion, though I think this will help us reduce errors in defining kfdefs.

Reply:
Good point.